build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/extractor/GoPublish.java

```diff
@@ -0,0 +1,179 @@
+        // We use the same code as Artifactory when he resolve a Go module directly from Github.
+        String deploayableZipPath = path.toString() + "/" + LOCAL_DEPLOYABLE_ZIP_FILENAME;
+        File deployableZipFile = new File(deploayableZipPath);
+        GoZipBallStreamer pkgArchiver = new GoZipBallStreamer(new ZipFile(tmpZipFile), moduleName, version, logger);
+        pkgArchiver.writeDeployableZip(deployableZipFile);
+
+        Artifact deployedPackage = deploy(deployableZipFile, PKG_ZIP_FILE_EXTENSION);
+        artifactList.add(deployedPackage);
+
+        // After uploading the package zip we can delete both zip files.
+        tmpZipFile.delete();
+        deployableZipFile.delete();
+    }
+
+    private void deployGoMod() throws IOException, NoSuchAlgorithmException {
+        String modLocalPath = getModFilePath();
+        Artifact deployedMod = deploy(new File(modLocalPath), PKG_MOD_FILE_EXTENSION);
+        artifactList.add(deployedMod);
+    }
+
+    private void createAndDeployInfo() throws IOException, NoSuchAlgorithmException {
+        String infoLocalPath = path.toString() + "/" + LOCAL_INFO_FILENAME;
+        File infoFile = writeInfoFile(infoLocalPath);
+        Artifact deployedInfo = deploy(infoFile, PKG_INFO_FILE_EXTENSION);
+        artifactList.add(deployedInfo);
+        infoFile.delete();
+    }
+
+    private File archiveProjectDir(String localZipPath) throws IOException {
+        File zipFile = new File(localZipPath);
+
+        try (ZipOutputStream zos = new ZipOutputStream(new FileOutputStream(zipFile))) {
+            List<Path> pathsList = Files.walk(path)
+                    .filter(p -> !Files.isDirectory(p) && !p.toFile().getName().equals(LOCAL_TMP_ZIP_FILENAME))
+                    .collect(Collectors.toList());
+            for (Path filePath : pathsList) {
+                ZipEntry zipEntry = new ZipEntry(path.relativize(filePath).toString());
+                zos.putNextEntry(zipEntry);
+                Files.copy(filePath, zos);
+                zos.closeEntry();
+            }
+        }
+
+        return zipFile;
+    }
+
+    /*
+     * pkg info is a json file contains:
+     *      1. The package's version.
+     *      2. The package creation timestamp.
+     */
+    private File writeInfoFile(String localInfoPath) throws IOException{
+        File infoFile = new File(localInfoPath);
+        ObjectMapper mapper = new ObjectMapper();
+        Map<String, String> infoMap = new HashMap();
+        Date date = new Date();
+        Instant instant = date.toInstant();
+
+        infoMap.put("Version", version);
+        infoMap.put("Time", instant.toString());
+
+        mapper.writeValue(infoFile, infoMap);
+        return infoFile;
+    }
+
+    /* Deploy pkg file and add it as an buildInfo's artifact */
+    private Artifact deploy(File deployedFile, String extension) throws IOException, NoSuchAlgorithmException {
```

Comment:
No need to declare two exception types thrown by this method, because eventually, you are catching Exception.
You can therefore declare this method as throwing Exception and by that, also avoid re-declaring these two exception types by the upstream methods.